### PR TITLE
fix: update utils image reference to latest image

### PR DESCRIPTION
**Pull request by mmalina**
The simple-e2e was failing because the tag no longer exists after David did a cleanup of a lot of older tags.

### CODE REVIEW

**Review by johnbieren — APPROVED** [review]

**Review by seanconroy2021 — APPROVED** [review]

File: integration-tests/pipelines/simple-e2e-pipeline.yaml [hunk 1]
```diff
@@ -84,7 +84,7 @@ spec:
             type: string
         steps:
           - name: check-if-test-relevant
-            image: quay.io/konflux-ci/release-service-utils:56d2d170ccaf0db49a49df3b978c602ee8d9e0fa
+            image: quay.io/konflux-ci/release-service-utils@sha256:14cc7c9550cf0cecf0ba675e6ac827994f183391b94cba5a1c9a63dce5ef4f67
```
[the remark this answers is not in the document]
Comment (mmalina) [reply integration-tests/pipelines/simple-e2e-pipeline.yaml:87]: Maybe in the future...

**Comment by mmalina** [comment]
The test failed on [create-internalrequest-on-remote-cluster](https://konflux-ui.apps.stone-prd-rh01.pg1f.p1.openshiftapps.com/ns/rhtap-release-2-tenant/pipelinerun/internal-services-simple-e2e-test-hklfs/logs/create-internalrequest-on-remote-cluster). Seems unrelated. I cannot load the logs ATM, I am getting Gateway timeout.

**Comment by mmalina** [comment]
It failed similarly again. I discussed it with the team in Slack. It seems it's just a timing issue that can happen sometimes.

**Comment by mmalina** [comment]
Wait, I checked the latest failed run in the merge queue and it [failed](https://konflux-ui.apps.stone-prd-rh01.pg1f.p1.openshiftapps.com/ns/rhtap-release-2-tenant/applications/internal-services/pipelineruns/internal-services-simple-e2e-test-v9lrq/) on the image pull:
```
the step "check-if-test-relevant" in TaskRun "internal-services-simple-e2e-test-v9lrq-check-if-test-relevant" failed to pull the image "". The pod errored with the message: "Back-off pulling image "quay.io/konflux-ci/release-service-utils:56d2d170ccaf0db49a49df3b978c602ee8d9e0fa": ErrImagePull: initializing source docker://quay.io/konflux-ci/release-service-utils:56d2d170ccaf0db49a49df3b978c602ee8d9e0fa: reading manifest 56d2d170ccaf0db49a49df3b978c602ee8d9e0fa in quay.io/konflux-ci/release-service-utils: unknown: Tag 56d2d170ccaf0db49a49df3b978c602ee8d9e0fa was deleted or has expired. To pull, revive via time machine."
```

How is that possible?

**Comment by johnbieren** [comment]
> Wait, I checked the latest failed run in the merge queue and it [failed](https://konflux-ui.apps.stone-prd-rh01.pg1f.p1.openshiftapps.com/ns/rhtap-release-2-tenant/applications/internal-services/pipelineruns/internal-services-simple-e2e-test-v9lrq/) on the image pull:
> 
> ```
> the step "check-if-test-relevant" in TaskRun "internal-services-simple-e2e-test-v9lrq-check-if-test-relevant" failed to pull the image "". The pod errored with the message: "Back-off pulling image "quay.io/konflux-ci/release-service-utils:56d2d170ccaf0db49a49df3b978c602ee8d9e0fa": ErrImagePull: initializing source docker://quay.io/konflux-ci/release-service-utils:56d2d170ccaf0db49a49df3b978c602ee8d9e0fa: reading manifest 56d2d170ccaf0db49a49df3b978c602ee8d9e0fa in quay.io/konflux-ci/release-service-utils: unknown: Tag 56d2d170ccaf0db49a49df3b978c602ee8d9e0fa was deleted or has expired. To pull, revive via time machine."
> ```
> 
> How is that possible?

does the merge queue e2e run use the pipeline definition from main or something?

**Comment by mmalina** [comment]
> > Wait, I checked the latest failed run in the merge queue and it [failed](https://konflux-ui.apps.stone-prd-rh01.pg1f.p1.openshiftapps.com/ns/rhtap-release-2-tenant/applications/internal-services/pipelineruns/internal-services-simple-e2e-test-v9lrq/) on the image pull:
> > ```
> > the step "check-if-test-relevant" in TaskRun "internal-services-simple-e2e-test-v9lrq-check-if-test-relevant" failed to pull the image "". The pod errored with the message: "Back-off pulling image "quay.io/konflux-ci/release-service-utils:56d2d170ccaf0db49a49df3b978c602ee8d9e0fa": ErrImagePull: initializing source docker://quay.io/konflux-ci/release-service-utils:56d2d170ccaf0db49a49df3b978c602ee8d9e0fa: reading manifest 56d2d170ccaf0db49a49df3b978c602ee8d9e0fa in quay.io/konflux-ci/release-service-utils: unknown: Tag 56d2d170ccaf0db49a49df3b978c602ee8d9e0fa was deleted or has expired. To pull, revive via time machine."
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > How is that possible?
> 
> does the merge queue e2e run use the pipeline definition from main or something?

Yeah, I think I got to the bottom of it. The gist of it is that integration tests don't really support merge queues. And so the event that triggers the build in the merge queue is in fact a push event. And because of this, the main reference of the e2e pipeline in this repo stays hardcoded to main. The actual Snapshot is fresh - so if this was a code fix, it would be fine. But the pipeline referenced in the ITS will be from main in the merge queue. See this comment in every ITS we have:
```
  contexts:
    - description: execute the integration test for a Snapshot created for a `pull request` event
      name: pull_request
      # This is a workaround for the fact that merge_group events are not reacted on by the
      # integration service
    - description: execute the integration test for a Snapshot created for a `push` event
      name: push
```

So the only way to merge this is to bypass / disable merge queues. Feel free to do it now, or I will pick it up tomorrow.

**Comment by mmalina** [comment]
Full explanation from Claude which I think is spot on:

```
That comment in the ITS file is the key — "workaround for the fact that merge_group events are not
  reacted on by the integration service." That tells me exactly what's going on.

  The actual mechanism

  Konflux's integration-service normally overrides an IntegrationTestScenario's git-resolver revision
  param at test-run time to match the Snapshot's own source commit — that's why your PR run
  correctly picked up commit 91443f1 (with your fix) even though the ITS has revision: main
  hardcoded.

  But real GitHub merge-queue runs fire a merge_group event, which integration-service doesn't
  understand at all. So this repo's .tekton/internal-services-pull-request.yaml fakes it: it matches
  merge-queue commits as a plain push event (target_branch.startsWith("gh-readonly-queue/main/")),
  and the ITS declares a push context to handle it.

  The problem: for a real push event (an actual merge landing on main), "use revision: main" is
  correct, because by the time that push-triggered test runs, the pushed commit is main's HEAD — so
  no override is needed. Konflux relies on that assumption and doesn't rewrite revision for
  push-context Snapshots.

  But a merge-queue "push" is not a real push — it's a tentative merge commit sitting on
  gh-readonly-queue/main/pr-906-..., and real main hasn't been updated yet. Since the push-context
  ITS never gets its revision overridden, it dutifully fetches
  integration-tests/pipelines/simple-e2e-pipeline.yaml from actual main — which still has the deleted
  tag, because your fix isn't merged yet.
```

and

```
  Two separate things get fetched, from two separate places

  1. The controller image under test — built by .tekton/internal-services-pull-request.yaml from the
  actual commit being tested (PR head, or the merge-queue's tentative merge commit). This is always
  fresh; it's never pinned to main. The result is packaged into a Snapshot that gets passed into the
  e2e test via the SNAPSHOT param (integration-tests/pipelines/simple-e2e-pipeline.yaml:11) — that's
  what the test actually exercises.
  2. The e2e pipeline definition itself (the YAML file that orchestrates the test, i.e.
  integration-tests/pipelines/simple-e2e-pipeline.yaml) — fetched by the ITS's git resolver, which
  for merge-queue ("push"-context) runs is pinned to literal main, as we found.

  These are independent. A normal application bug fix (e.g. something in controllers/ or api/)
  changes what goes into artifact #1 — the freshly-built image, referenced by digest in the Snapshot.
  The test harness (#2) just runs whatever steps are in the (possibly stale) pipeline file against
  that fresh image. Since the harness's logic ("build, deploy, run these checks") doesn't itself need
  to change for a typical code fix, reading a slightly-stale-but-functionally-identical copy of the
  test file from main doesn't matter — it still correctly tests your new image.

  The staleness only bites when the fix you're merging is a change to the test harness file itself —
  exactly your case, where the broken thing was a hardcoded image reference inside
  simple-e2e-pipeline.yaml. That file is self-referential: it's both the thing you're fixing and the
  thing whose fetch is pinned to main. That's a fairly rare edge case (test-infra fixes), not the
  general case (application code fixes), so it's consistent with this not being a routinely-hit
  blocker for the team.

  So: your working theory is right, and the fix is exactly what we discussed — get this specific fix
  onto main via a merge-queue bypass once, and normal operation resumes since future PRs/merge-queue
  runs will read the corrected simple-e2e-pipeline.yaml from main.
```
